File: Creational_Patterns/Factory_Pattern/python/Basic.py
from abc import ABC, abstractmethod
from typing import override

class Shape(ABC):
  @abstractmethod
  def draw(self):
    pass

class Circle(Shape):
  @override
  def draw(self):
    print("Drawing Circle")


class Square(Shape):
  @override 
  def draw(self):
    print("Drawing Square")
    

class ShapeFactory:
  def get_shape(self, shape_type):
    if shape_type.lower() == 'circle':
      return Circle()
    elif shape_type.lower() == 'sqaure':
      return Square()
    return None


shape_factory = ShapeFactory()

circle = shape_factory.get_shape("Circle")
square = shape_factory.get_shape("Sqaure")


circle.draw()
square.draw()

File: Creational_Patterns/Factory_Pattern/java/Basic.java
interface Shape{
  public void draw();
}

class Circle implements Shape{
  @Override
  public void draw(){
    System.out.println("Drawing Circle");
  }
}


class Square implements Shape{
  @Override
  public void draw(){
    System.out.println("Drawing Square");
  }
}


class ShapeFactory{
  public static Shape getShape(String shape){
    if(shape.equalsIgnoreCase("circle")){
      return new Circle();
    }else if(shape.equalsIgnoreCase("square")){
      return new Square();
    }else{
      return null;
    }
  }
}


class Main{
  public static void main(String args[]){
    Shape circle = ShapeFactory.getShape("circle");
    Shape square = ShapeFactory.getShape("square");

    circle.draw();
    square.draw();
  }
}